src/main.c

```
@@ -673,6 +683,57 @@ int main(s32 argc, char *argv[]) {
 		case ID_RELEASE:
 			opt.target->opt = ASSEMBLY_OPT_RELEASE_FAST;
 			break;
+		case ID_INIT_PROJECT:
+			if (file_exists("build.bl")) {
+				builder_error("File 'build.bl' exists in the current directory.");
+				EXIT(EXIT_FAILURE);
+			}
+			char *project_name = "out";
+			if (argv[index + 1]) {
+				project_name = argv[index + 1];
+			}
+
+			FILE *build_file = fopen("build.bl", "w+");
+			if (!build_file) {
+				builder_error("could not create build file!");
+				EXIT(EXIT_FAILURE);
+			}
+			const char *build_function_code_template = "build :: fn () #build_entry {\n"
+			                                           "    exe := add_executable(\"%s\");\n"
+			                                           "    set_output_dir(exe,\"bin\");\n"
+			                                           "    add_unit(exe, \"src/main.bl\");\n"
+			                                           "    compile(exe);\n"
+			                                           "}\n";
+
+			char *exe_name = project_name;
+
+			fprintf(build_file, "\n\n\n");
+			fprintf(build_file, build_function_code_template, exe_name);
+
+			if (!create_dir("bin")) {
+				EXIT(EXIT_FAILURE);
```

Please add an error report instead of a silent failure. Same for this `src` creation.